scapy/main.py

```diff
@@ -65,26 +68,33 @@ def _usage():
 ######################
 
 
-def _load(module):
+def _load(module, globals_dict=None, symb_list=None):
+    if globals_dict is None:
+        globals_dict = six.moves.builtins.__dict__
     try:
         mod = importlib.import_module(module)
         if '__all__' in mod.__dict__:
             # import listed symbols
             for name in mod.__dict__['__all__']:
-                six.moves.builtins.__dict__[name] = mod.__dict__[name]
+                if symb_list is not None:
+                    symb_list.append(name)
+                globals_dict[name] = mod.__dict__[name]
         else:
             # only import non-private symbols
             for name, sym in six.iteritems(mod.__dict__):
                 if _validate_local(name):
-                    six.moves.builtins.__dict__[name] = sym
-    except Exception as e:
-        log_interactive.error(e)
+                    if symb_list is not None:
+                        symb_list.append(name)
+                    globals_dict[name] = sym
+    except Exception:
+        log_interactive.error("Loading module %s", module, exc_info=True)
 
 def load_module(name):
     _load("scapy.modules."+name)
 
-def load_layer(name):
-    _load("scapy.layers." + LAYER_ALIASES.get(name, name))
+def load_layer(name, globals_dict=None, symb_list=None):
```

Comment:
What do you think of ?
`"""Loads a Scapy layers to make variables, objects and functions available globally"""`